src/data/matrix/basic.lean

```diff
@@ -76,6 +77,13 @@ lemma map_add [add_monoid α] {β : Type w} [add_monoid β] (f : α →+ β)
   (M N : matrix m n α) : (M + N).map f = M.map f + N.map f :=
 by { ext, simp, }
 
+lemma map_sub [add_group α] {β : Type w} [add_group β] (f : α →+ β)
+  (M N : matrix m n α) : (M - N).map f = M.map f - N.map f :=
+by { ext, simp }
+
+lemma subsingleton_of_empty (hn : ¬ nonempty n) : subsingleton (matrix n n α) :=
```

Comment:
Could you please split this into `subsingleton_of_empty_index_left` and `subsingleton_of_empty_index_right` and prove it for `matrix m n \a` (i.e. include non-square matrices)?